code/datums/gamemode/objectives/target/locate.dm

```diff
@@ -0,0 +1,49 @@
+/datum/objective/target/locate
+	name = "locate objects"
+	var/list/objects_to_locate = list()
+
+/datum/objective/target/locate/IsFulfilled()
+	.=..()
+	if(objects_to_locate.len)
+		return FALSE
+	return TRUE
+
+/datum/objective/target/locate/random/find_target()
+	var/list/potential_objects = list(/obj/item/weapon/bikehorn/rubberducky,
+			/obj/item/weapon/hand_tele,
+			/obj/item/weapon/gun/energy/laser/captain,
+			/obj/item/weapon/disk/nuclear,
+			/obj/item/weapon/reagent_containers/glass/bucket,
+			)
+	for(var/i = 0 to rand(3,potential_objects.len-1))
+		var/pick = pick(potential_objects)
+		objects_to_locate.Add(pick)
+		potential_objects.Remove(pick)
+	explanation_text = format_explanation()
+	return TRUE
+
+/datum/objective/target/locate/format_explanation()
+	var/explanation = "Locate "
+	if(objects_to_locate.len)
+		if(objects_to_locate.len > 1)
+			for(var/i in objects_to_locate)
+				var/obj/item/I = i
+				if(objects_to_locate[i] != objects_to_locate[objects_to_locate.len-1])
+					explanation += "a [initial(I.name)], "
+				else
+					explanation += "& a [initial(I.name)]."
+		else
+			var/obj/item/I = objects_to_locate[1]
+			explanation += " a [initial(I.name)]."
+	else
+		explanation = "All items located."
+	return explanation
+
+/datum/objective/target/locate/proc/check(var/list/objects)
```

Comment:
objective feels slightly easy since you only need to see it once, not gather them all to one place

I feel like it would be more thematic if you had to gather them together, like
>get a picture of a janitor, a barsign, and runtime - this is an event that has to happen

The means you may need to kidnap someone, steal something, and set up an event in a very specific area.

Then again, maybe this harder version of the objective might just be better for an alt objective like the assassination one.